### PR TITLE
OpsWorks Services Clarification

### PR DESCRIPTION
**Pull request by ncalteen**
Updating OpsWorks information:
- Removing unclear statements regarding usage.
- Clarify different OpsWorks services (Stacks, Chef Automate, Puppet Enterprise).
- Clarify Elastic Beanstalk is not a configuration management service (EB is an orchestration service/PaaS).
- Clarify usage of lifecycle events in OpsWorks Stacks (and that Chef runs can be manually triggered).
- Updated service availability (clarified by OpsWorks services).

### CODE REVIEW

**Comment by ncalteen** [comment]
Per your contribution guidelines, I'd like to mention my affiliation. I am an AWS employee and OpsWorks SME.

File: README.md [hunk 1]
```diff
@@ -2032,31 +2032,43 @@ OpsWorks
...
-- [OpsWorks Stacks](https://aws.amazon.com/opsworks/stacks/) allows you to run your deployment stacks both in the AWS cloud as well as on your own hardware on-premises.
-- [OpsWorks for ChefAutomate](https://aws.amazon.com/opsworks/chefautomate/) provides a managed Chef configuration management server for your deployment pipeline. This server stores configuration tasks and provides them to your deployment nodes without manual intervention, in addition to providing other management and monitoring features.
+- OpsWorks is a configuration management service that uses [Chef](https://www.chef.io/chef/) or [Puppet](https://www.puppet.com) configuration management. It is broken out into three different services:
+  - OpsWorks Stacks: The service lets you configure and launch stacks specific to your application's needs, and allows you to automate application deployments. Chef runs can be performed manually via the Execute Cookbooks command, otherwise they are only run as part of lifecycle events.
```
Comment (DavidTPate) [inline README.md]: Could we add the links to the individual pages back? ie. `[OpsWorks Stacks](https://aws.amazon.com/opsworks/stacks/)`.

Also, Stacks support running on any target system ie. EC2 or on-prem. I think that's a useful note here.

File: README.md [hunk 2]
```diff
@@ -2032,31 +2032,43 @@ OpsWorks
...
+  - OpsWorks for Chef Automate: This service launches a dedicated Chef Automate server in your account, which can be used to associate notes, upload coobook code, and configure systems. Automated patching, backups, OS updates, and minor Chef version upgrades are provided as part of the service. An AWS API is provided for associating/disassociating nodes. Chef runs can be scheduled on nodes using the [chef-client cookbook](https://supermarket.chef.io/cookbooks/chef-client).
+  - OpsWorks for Puppet Enterprise: This service launches a dedicated Puppet Master in your account, which can be used to assocaite notes, upload modules, and configure systems. Automated patching, backups, OS updates, and minor Puppet version upgrades are provided as part of the service. An AWS API is provided for associating/disassociating nodes. By default, the Puppet agent will run automatically every 30 minutes on associated nodes.
```
Comment (DavidTPate) [inline README.md]: Mispelling on `associate notes`

File: README.md [hunk 3]
```diff
@@ -2032,31 +2032,43 @@ OpsWorks
...
+- OpsWorks Stacks differs from standard configuration management services in that it also allows you to perform some infrastructure and application automation (such as creating Amazon EC2 instances and deploying applications via Chef cookbooks).
```
Comment (DavidTPate) [inline README.md]: Seems like this should be up before line 2037.

File: README.md [hunk 4]
```diff
@@ -2032,31 +2032,43 @@ OpsWorks
...
 
 ### OpsWorks Tips
 
-- [OpsWorks relies heavily on Chef cookbooks and recipes](http://docs.aws.amazon.com/opsworks/latest/userguide/gettingstarted-cookbooks.html) for customization, so familiarity with reading their syntax will help greatly with getting up and running.
+- OpsWorks Stacks and OpsWorks for Chef Automate use Chef cookbooks for configuration. Chef provides free training to learn syntax, best practices, etc. at [https://learn.chef.io](https://learn.chef.io).
+- OpsWorks for Puppet Enterprise uses Puppet manifests for configuration. Puppet provides a very useful learning VM for download at [https://learn.puppet.com/](https://learn.puppet.com/).
 
 ### OpsWorks Gotchas and Limitations
 
-- Although OpsWorks will let you work with common Chef recipes or Puppet modules when creating your stacks, creating custom recipes will require familiarity with Chef or Puppet syntax.
```
Comment (DavidTPate) [inline README.md]: This still seems to be a relevant comment.

File: README.md (hunk 1 above)
Comment (ncalteen) [reply README.md]: Sounds good. I would clarify that all three services support managing Amazon EC2 or on-premises infrastructure, though the implementation details differ.

File: README.md (hunk 4 above)
Comment (QuinnyPig) [reply README.md]: Yeah, let's keep this one in please. The rest looks great!

Comment (ncalteen) [reply README.md]: New revision has been posted with this included.

Comment (QuinnyPig) [reply README.md]: I'm not seeing this in the diff. Am I going blind? :-)

Comment (ncalteen) [reply README.md]: Sorry! Reworded in the last commit to include scope of AWS support.

“Although OpsWorks will let you work with common Chef recipes or Puppet modules when creating your stacks, creating custom recipes will require familiarity with Chef or Puppet syntax. Chef/Puppet code is not supported as part of AWS Support.”

**Comment by QuinnyPig** [comment]
That got it-- thanks! Merging; we appreciate the help!

**Pull request merged**